libraries/azure_generic_resource.rb

```diff
@@ -21,8 +21,12 @@ def initialize(opts = {}, static_resource = false)
       validate_generic_resource
       return if failed_resource?
     end
-    @display_name = @opts.slice(:resource_group, :resource_provider, :name, :tag_name, :tag_value, :resource_id,
-                                :resource_uri).values.join(' ') if @opts[:display_name].nil?
+    if @opts[:display_name].nil?
+      @display_name = @opts.slice(:resource_group, :resource_provider, :name, :tag_name, :tag_value, :resource_id,
+                                  :resource_uri).values.join(' ')
+    else
+      @display_name = @opts[:display_name]
+    end
```

Comment:
I think you can do something along these lines...
```
@display_name = if @opts[:display_name].nil?
                  @opts.slice(:resource_group, :resource_provider, :name, :tag_name, :tag_value, :resource_id, :resource_uri).values.join(' ')
                else
                  @opts[:display_name]
                end
```

Not necessary to change. Up to you which you prefer